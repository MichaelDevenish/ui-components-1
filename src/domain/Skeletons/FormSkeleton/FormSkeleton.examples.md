**Form Skeleton** shows a form-like skeleton

It supports 1 column and 2 column form styles 

```jsx
  <FormSkeleton />
```

### Best Practices
* Match the skeleton styling to what your form actually looks like

<br />

Two Column Variant:

```jsx
  <FormSkeleton variant={FormSkeleton.Variant.TwoColumn} />
```

The `showButtonSkeleton` is useful for displaying skeletons for forms in Modals
and other cases places where the submit button will not sit on the bottom left

```jsx
  <FormSkeleton showButtonSkeleton={false} />
```
