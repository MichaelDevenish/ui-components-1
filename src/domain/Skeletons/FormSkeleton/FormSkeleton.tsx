import React from 'react'

import { Variables } from '../../../common'
import { BlockSkeleton } from '../BlockSkeleton'

const FormSkeleton: React.FC = () => {
    return (
      <>
        <FormFieldSkeleton />
        <FormFieldSkeleton />
        <FormFieldSkeleton />
        <BlockSkeleton
          width={120}
          height={40}
          margins={{ top: Variables.Spacing.sXSmall }}
        />
      </>
    )
}

const FormFieldSkeleton: React.FC = () => {
  return (
    <>
      <BlockSkeleton
        width={200}
        height={24}
        margins={{ bottom: Variables.Spacing.sXSmall }}
      />
      <BlockSkeleton
        height={40}
        margins={{ bottom: Variables.Spacing.sLarge }}
      />
    </>
  )
}

export {
  FormSkeleton,
  FormFieldSkeleton
}
