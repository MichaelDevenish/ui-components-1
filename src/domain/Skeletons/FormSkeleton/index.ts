export { FormFieldSkeleton, FormSkeleton } from './FormSkeleton'
