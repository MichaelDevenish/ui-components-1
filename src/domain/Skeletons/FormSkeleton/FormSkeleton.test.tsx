import { shallow } from 'enzyme'
import React from 'react'

import { FormFieldSkeleton, FormSkeleton } from './FormSkeleton'

describe('<FormSkeleton />', () => {
  it('should render a form skeleton', () => {
    const wrapper = shallow(
      <FormSkeleton />
    )

    expect(wrapper).toMatchSnapshot()
  })
})

describe('<FormFieldSkeleton />', () => {
  it('should render a form field skeleton', () => {
    const wrapper = shallow(
      <FormFieldSkeleton />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
