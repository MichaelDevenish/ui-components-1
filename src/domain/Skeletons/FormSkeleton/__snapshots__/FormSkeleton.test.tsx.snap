// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormFieldSkeleton /> should render a form field skeleton 1`] = `
<Fragment>
  <BlockSkeleton
    display="block"
    height={24}
    margins={
      Object {
        "bottom": 8,
      }
    }
    showSkeleton={true}
    width={200}
  />
  <BlockSkeleton
    display="block"
    height={40}
    margins={
      Object {
        "bottom": 24,
      }
    }
    showSkeleton={true}
  />
</Fragment>
`;

exports[`<FormSkeleton /> should render a form skeleton 1`] = `
<Fragment>
  <FormFieldSkeleton />
  <FormFieldSkeleton />
  <FormFieldSkeleton />
  <BlockSkeleton
    display="block"
    height={40}
    margins={
      Object {
        "top": 8,
      }
    }
    showSkeleton={true}
    width={120}
  />
</Fragment>
`;
